setup.py

```diff
@@ -128,9 +128,35 @@ def run(self):
 
 version = '0.0.6'  # preserve format, this is read from __init__.py
 
+
+def read_version_from_git():
+    try:
+        import subprocess
+        import shlex
+        git_version, err = subprocess.Popen(
+            shlex.split('git describe --tags'),
+            stdout=subprocess.PIPE,
+        ).communicate()
+        if git_version.startswith('v'):
+            git_version = git_version[1:]
+
+        git_version = git_version.strip()
+        # if this is has commits after the tag, it's a prerelease:
+        if git_version.count('-') == 2:
+            spec, _, commit = git_version.split('-')
+            if commit.startswith('g'):
+                commit = commit[1:]
+            return '{}+git.r{}'.format(spec, commit)
+        else:
+            return git_version
+    except BaseException as e:
+        print('could not read version from git: {}'.format(e))
```

Comment:
Do we want to have this here? It would show up on every installation that is not from a git checkout (which we currently don't officially support, but still)